Standalone/PythonTests/Automated/benchmark_data_aggregator.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+"""
+Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+from argparse import ArgumentParser
+import json
+from pathlib import Path
+import time
+import subprocess
+
+from ly_test_tools.mars.filebeat_client import FilebeatClient
+
+class BenchmarkPathException(Exception):
+    """Custom Exception class for invalid benchmark file paths."""
+    pass
+
+class BenchmarkDataAggregator(object):
+    def __init__(self, workspace, logger, test_suite):
+        self.build_dir = workspace.paths.build_directory()
+        self.results_dir = Path(workspace.paths.project(), 'user/Scripts/PerformanceBenchmarks')
+        self.test_suite = test_suite
+        self.filebeat_client = FilebeatClient(logger)
+
+    def _process_benchmark(self, benchmark_dir, benchmark_metadata):
+        '''
+        Aggregates data from results from a single benchmark contained in a subdirectory of
+        self.results_dir.
+        :param benchmark_dir: Path of directory containing the benchmark results
+        :param benchmark_metadata: Base object for metadata associated with the benchmark
+        :return: A list of tuples representing the Elasticsearch index suffix associated with
+                 the payload to deliver to Elasticsearch via Filebeat
+        '''
+        # Parse benchmark metadata
+        metadata_file = benchmark_dir / 'benchmark_metadata.json'
+        if metadata_file.exists():
+            data = json.loads(metadata_file.read_text())
+            benchmark_metadata.update(data['ClassData'])
+        else:
+            raise BenchmarkPathException(f'Metadata file could not be found at {metadata_file}')
+
+        # data structures aggregating statistics from timestamp logs
+        frame_stats = { 'count': 0, 'totalTime': 0, 'maxTime': 0, 'minTime': float('inf') }
+        pass_stats = {}  # key: pass name, value: dict with totalTime and maxTime keys
+
+        # this allows us to add additional data if necessary, e.g. frame_test_timestamps.json
+        is_timestamp_file = lambda file: file.name.startswith('frame') and file.name.endswith('_timestamps.json')
+        def update_pass(entry):
```

Comment:
instead of nesting this function inside `_process_benchmark()`, can we just pass in the values needed from the nesting in as params? i.e. `def update_pass(entry, pass_stats):`

that way we won't need to nest it inside `_process_benchmark()` and it can exist outside of it and also make it callable by other functions should that ever be desired in a future update